core/src/main/java/dk/alexandra/fresco/framework/util/HmacDrbg.java

```diff
@@ -0,0 +1,143 @@
+package dk.alexandra.fresco.framework.util;
+
+import dk.alexandra.fresco.framework.MPCException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.crypto.Mac;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements the HMac-DRBG (HMac-Deterministic Random Bit Generator) as specified in
+ * https://csrc.nist.gov/publications/detail/sp/800-90a/rev-1/final
+ *
+ * <p>
+ * Note that this class is not threadsafe.
+ * </p>
+ * <p>
+ * Note also that reseeding (calling {@link #setSeed(byte[])}) should occur at least every 2^42
+ * calls to {@link #nextBytes(byte[])} to be secure. No signal will be send about this.
+ * </p>
+ */
+public class HmacDrbg implements Drbg {
+
+  private final Logger logger = LoggerFactory.getLogger(HmacDrbg.class);
+  static String DEFAULT_ALGORITHM = "HmacSHA256";
+  private final String algorithm;
+  private Mac mac = null;
+  private byte[] val = null; // value - internally used
+  private byte[] key = null; // key material
+  long reseedCounter;
+  private List<byte[]> seeds;
+  static final long MAX_RESEED_COUNT = (long) Math.pow(2, 48);
```

Comment:
Wouldn't `1L << 48` be nicer here?

Reply:
No, I don't think so. It is not as obvious what the number means, and performance-wise it doesn't really matter.

Reply:
I beg to differ, 1 << 48 is clear to me, whereas Math.pow is more arithmetic and computation oriented.

What really bugs is the double, I would definitely expect a long and no casts, we are talking about a counter maximum here.

Reply:
It's not about what is clear to you, it is about what is clear to the general population using the code. I can just change it, no problem, but I still think more ppl would understand the power construct. I can use a BigInteger construct if you prefer this over double?

Reply:
Meh.. Nevermind. I'll just do the << and add a comment. np..